src/ngraph_backend_manager.cc

```diff
@@ -107,7 +109,28 @@ bool BackendManager::IsSupportedBackend(const string& backend_name) {
     return false;
   }
   return true;
-};
+}
+
+Status BackendManager::GetCurrentlySetBackendName(string* current_backend) {
+  string possible_backend;
+  const char* ng_backend_env_value = std::getenv("NGRAPH_TF_BACKEND");
+  // Environment variable has higher priority than API set value
+  if (ng_backend_env_value != nullptr) {
+    possible_backend = std::string(ng_backend_env_value);
+  } else {
+    possible_backend = BackendManager::ng_backend_name_;
+  }
+  // Check if possible_backend is valid
+  if (possible_backend.empty() ||
+      !BackendManager::IsSupportedBackend(possible_backend)) {
+    current_backend = nullptr;
+    return errors::Internal("NGRAPH_TF_BACKEND: ", possible_backend,
+                            " is not supported");
+  } else {
+    *current_backend = possible_backend;
+    return SetBackendName(*current_backend);
```

Comment:
This is tricky. The API says GetCurrentlySetBackendName(), and we also SetBackendName() in it. It might be ok to not SetBackend Here.

Reply:
yes probably should not set in get

We can discuss this scenario on monday:


initially backend='cpu'
env var is set to GPU
someone calls getbackend. it tries to return cpu, but sees envvar = gpu, so returns GPU

now env var is unset

next time someone queries getbackend, it would return "cpu". if i had called set in the get earlier it would return "gpu"
We want to decide if it should return cpu or gpu at this point.

if think we should return "cpu" (hence there should be no set in get)